gdal/ogr/ogrpolyhedralsurface.cpp

```diff
@@ -0,0 +1,857 @@
+    for( int iGeom = 0; iGeom < oMP.nGeomCount; iGeom++ )
+    {
+        oMP.papoGeoms[iGeom]->exportToWkb( eByteOrder, pabyData + nOffset, eWkbVariant );
+        nOffset += oMP.papoGeoms[iGeom]->WkbSize();
+    }
+
+    return OGRERR_NONE;
+}
+
+/************************************************************************/
+/*                           importFromWkt()                            */
+/*              Instantiate from well known text format.                */
+/************************************************************************/
+
+OGRErr OGRPolyhedralSurface::importFromWkt( char ** ppszInput )
+
+{
+    int bHasZ = FALSE, bHasM = FALSE;
+    bool bIsEmpty = false;
+    OGRErr      eErr = importPreambuleFromWkt(ppszInput, &bHasZ, &bHasM, &bIsEmpty);
+    flags = 0;
+    if( eErr != OGRERR_NONE )
+        return eErr;
+    if( bHasZ ) flags |= OGR_G_3D;
+    if( bHasM ) flags |= OGR_G_MEASURED;
+    if( bIsEmpty )
+        return OGRERR_NONE;
+
+    char        szToken[OGR_WKT_TOKEN_MAX];
+    const char  *pszInput = *ppszInput;
+    eErr = OGRERR_NONE;
+
+    /* Skip first '(' */
+    pszInput = OGRWktReadToken( pszInput, szToken );
+
+/* ==================================================================== */
+/*      Read each surface in turn.  Note that we try to reuse the same  */
+/*      point list buffer from ring to ring to cut down on              */
+/*      allocate/deallocate overhead.                                   */
+/* ==================================================================== */
+    OGRRawPoint *paoPoints = NULL;
+    int          nMaxPoints = 0;
+    double      *padfZ = NULL;
+
+    do
+    {
+
+    /* -------------------------------------------------------------------- */
+    /*      Get the first token, which should be the geometry type.         */
+    /* -------------------------------------------------------------------- */
+        const char* pszInputBefore = pszInput;
+        pszInput = OGRWktReadToken( pszInput, szToken );
+
+        OGRSurface* poSurface;
+
+    /* -------------------------------------------------------------------- */
+    /*      Do the import.                                                  */
+    /* -------------------------------------------------------------------- */
+        if (EQUAL(szToken,"("))
+        {
+            OGRPolygon      *poPolygon = new OGRPolygon();
+            poSurface = poPolygon;
+            pszInput = pszInputBefore;
+            eErr = poPolygon->importFromWKTListOnly( (char**)&pszInput, bHasZ, bHasM,
+                                                     paoPoints, nMaxPoints, padfZ );
+        }
+        else if (EQUAL(szToken, "EMPTY") )
+        {
+            poSurface = new OGRPolygon();
+        }
+
+        /* We accept POLYGON() but this is an extension to the BNF, also */
+        /* accepted by PostGIS */
+        else if ((EQUAL(szToken,"POLYGON") ||
+                  EQUAL(szToken,"CURVEPOLYGON")))
```

Comment:
CURVEPOLYGON are not allowed in PolyhedralSurface, right ? So they shouldn't be accepted here